web-common/src/features/dashboards/state-managers/actions/core-actions.ts

```diff
@@ -1,11 +1,21 @@
 import { resetAllContextColumnWidths } from "./context-columns";
 import type { DashboardMutables } from "./types";
 
+// NOTE: this is a temporary action to set the leaderboard measure count.
+// It will be removed in the multiple measures second pass anyway.
 export const setLeaderboardMeasureCount = (
   { dashboard }: DashboardMutables,
   count: number,
 ) => {
   dashboard.leaderboardMeasureCount = count;
+
+  // If the current leaderboard measure is not in the first N visible measures,
+  // set it to the first visible measure
+  const visibleMeasures = dashboard.visibleMeasures.slice(0, count);
+  if (!visibleMeasures.includes(dashboard.leaderboardMeasureName)) {
+    dashboard.leaderboardMeasureName =
+      visibleMeasures[0] || dashboard.leaderboardMeasureName;
```

Comment:
I'm confused by the fallback `|| dashboard.leaderboardMeasureName`. What will happen in that case? Won't we always have a visible measure anyways?

Reply:
you're right, visibleMeasures[0] will always exist